test/unit/api/udt/udt.hpp

```diff
@@ -43,34 +35,17 @@ namespace udt
     if constexpr(I==0) return p.x; else if constexpr(I==1) return p.y;
   }
 
+  // Stream insertion is also on your behalf
   std::ostream& operator<<( std::ostream& os, grid2d const& p)
   {
     return os << "[x: " << p.x << " - y: " << p.y << "]";
   }
-
-  //------------------------------------------------------------------------------------------------
-  // Functions and operators are externally defined using tagged_dispatch
-  //------------------------------------------------------------------------------------------------
-  auto tagged_dispatch( eve::tag::is_equal_
-                      , eve::same_value_type<grid2d> auto a
-                      , eve::same_value_type<grid2d> auto b
-                      )
-  {
-    grid2d::eq_counter++;
-    return (get<0>(a) == get<0>(b)) &&(get<1>(a) == get<1>(b));
-  }
-
-  auto tagged_dispatch( eve::tag::is_not_equal_
-                      , eve::same_value_type<grid2d> auto a
-                      , eve::same_value_type<grid2d> auto b
-                      )
-  {
-    grid2d::neq_counter++;
-    return (get<0>(a) != get<0>(b)) || (get<1>(a) != get<1>(b));
-  }
 }
 
 // Opt-in for eve::product_type + adaptation
-template<>              struct eve::is_product_type<udt::grid2d>  : std::true_type {};
-template<>              struct std::tuple_size<udt::grid2d>       : std::integral_constant<std::size_t, 2> {};
-template<std::size_t I> struct std::tuple_element<I,udt::grid2d> { using type = int; };
+template<>              struct eve::is_product_type<udt::grid2d>    : std::true_type {};
+template<>              struct std::tuple_size<udt::grid2d>         : std::integral_constant<std::size_t, 2> {};
+template<std::size_t I> struct std::tuple_element<I,udt::grid2d>    { using type = int; };
+
+// Opt-in for SIMD < > <= >= operators
+template<>  struct eve::supports_ordering<udt::grid2d>  : std::true_type {};
```

Comment:
Can't we just like inherit smth or have an internal typedef for simplicity?

Reply:
If we force people to inherit from something, this means we can't adapt types we don't write. For ex., we could not adapt say QPoint or some OpenGL types.

Now, we ma have both, but it is important we have a non-intrusive option.

Reply:
Sure, but let's have an easy option too.

Reply:
The easy option is now that we have ordering by default. Other operators will have a dual traits/typedef system